abi-spec/core_test.go

```diff
@@ -0,0 +1,150 @@
+package abispec
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+)
+
+func TestMethodPattern(t *testing.T) {
+	results := methodPattern.FindStringSubmatch("sam(bytes,bool,uint256[])")
+	require.Equal(t, "sam(bytes,bool,uint256[])", results[0])
+	require.Equal(t, "sam", results[1])
+	require.Equal(t, "bytes,bool,uint256[]", results[2])
+
+	results = methodPattern.FindStringSubmatch("s(bytes,bool,uint256[])")
+	require.Equal(t, "s(bytes,bool,uint256[])", results[0])
+	require.Equal(t, "s", results[1])
+	require.Equal(t, "bytes,bool,uint256[]", results[2])
+
+	results = methodPattern.FindStringSubmatch("s1(bytes,bool,uint256[])")
+	require.Equal(t, "s1(bytes,bool,uint256[])", results[0])
+	require.Equal(t, "s1", results[1])
+	require.Equal(t, "bytes,bool,uint256[]", results[2])
+
+	results = methodPattern.FindStringSubmatch("1s(bytes,bool,uint256[])")
+	require.Len(t, results, 0)
+}
+
+//TODO to be fixed
```

Comment:
what should be fixed ?

Reply:
there is a corresponding [test case](https://github.com/status-im/status-mobile/blob/ca8b7211358e096bf91fa8c8d965a8855d8fe727/src/status_im/ethereum/abi_spec_test.cljs#L31) in status-mobile , but go-ethereum does not support arg type: `uint` ([test refference](https://github.com/ethereum/go-ethereum/blob/d10c28030944d1c32febba3f45ae8c175ab34063/accounts/abi/type_test.go#L126)), actually, uint is alias for uint256, i'm not sure if we need support `uint` in status-go ? @flexsurfer